tests/unit/clients/test_http.py

```diff
@@ -245,6 +245,36 @@ def client(mocker, auth_client):
     assert 2 == len(caplog.records)
 
 
+@pytest.mark.asyncio
+async def test_request_with_other_exception(
+        client, mocker, monkeypatch, caplog):
+    """Correct exception is raised when the aiohttp request itself fails."""
+
+    async def mock_refresh_token():
+        pass
+
+    monkeypatch.setattr(
+        client._auth_client, 'refresh_token', mock_refresh_token)
+
+    async def mock_valid_token_set():
+        pass
+
+    monkeypatch.setattr(client, 'valid_token_set', mock_valid_token_set)
+
+    mock_aiohttp_request = mocker.Mock()
+    mock_aiohttp_request.side_effect = ValueError  # randomly chosen
+    monkeypatch.setattr(
+        client._session, 'request', mock_aiohttp_request
```

Comment:
I thought this would be a coroutine?

Reply:
That's a good point.  I guess it works because you can await a non-awaitable and it just does nothing.  Maybe I should make an async function just for clarity?

Reply:
@econchick It turns out that doing it this way is actually much simpler.  If you make it a coro, you have to implement __aenter__ and __aexit__, while this way seems not to actually apply the context manager protocol.  Thoughts?

Reply:
It also turns out that the mock library won't let you mock a magic function like __aenter__.

Reply:
ftr mock doesn't support async stuff at all (maybe in 3.7? an unconfirmed rumor).

Reply:
OK, I've added a more thorough solution that actually also matches what aiohttp does better.

Reply:
@econchick re mock, I also discovered that if you try to add a method that starts with __, it won't let you.  Or maybe surrounded by __, didn't check the exact criteria.